engine/classes/Elgg/EntityIconService.php

```diff
@@ -225,9 +225,14 @@ public function saveIcon(ElggEntity $entity, ElggFile $file, $type = 'icon', arr
 		if ($created !== true) {
 			// remove existing icons
 			$this->deleteIcon($entity, $type, true);
-			
+
+			// save original image if 'original' icon_size is available for this type of entity
```

Comment:
Check if `original` and `master` are in the size config first. Attempt to create both, if either fails, delete.

Reply:
Thanks for the tip. I'll re-try when I have time. I might also have to adjust the tests. Not sure about that yet.